src/components/answerComponent.js

```diff
@@ -1,26 +1,35 @@
-/**
- * Create an Answer element
- * @returns {Element}
- */
-export const createAnswerComponent = (key, answerText) => {
+let clickHandlerFunctions = new Map();
+
+export const createAnswerComponent = (key, answerText, onSelect) => {
   const element = document.createElement('li');
   element.innerHTML = String.raw`
     <label for="${key}">
       <input type="radio" name="answerItem" value=${key} id="${key}" />
     ${answerText}
     </label>
   `;
-  element.addEventListener('click', (e) => {
+  const clickHandlerFunction = clickHandler(onSelect);
+  element.addEventListener('click', clickHandlerFunction);
+  clickHandlerFunctions.set(element, clickHandlerFunction);
+  return element;
+};
+
+const clickHandler = function (onSelect) {
```

Comment:
Make sure to define functions definition on top of function call!

Otherwise you will be depending on "Javascript hoisting" you can search about it.